src/main/java/com/iexec/commons/poco/order/RequestOrder.java

```diff
@@ -0,0 +1,137 @@
+import com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder;
+import com.iexec.commons.poco.contract.generated.IexecHubContract;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import lombok.Value;
+import org.web3j.utils.Numeric;
+
+import java.math.BigInteger;
+import java.util.Objects;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+@ToString(callSuper = true)
+@JsonDeserialize(builder = RequestOrder.RequestOrderBuilder.class)
+public class RequestOrder extends Order {
+
+    String app;
+    BigInteger appmaxprice;
+    String dataset;
+    BigInteger datasetmaxprice;
+    String workerpool;
+    BigInteger workerpoolmaxprice;
+    String requester;
+    BigInteger category;
+    BigInteger trust;
+    String beneficiary;
+    String callback;
+    String params;
+
+    @Builder
+    RequestOrder(
+            String app,
+            BigInteger appmaxprice,
+            String dataset,
+            BigInteger datasetmaxprice,
+            String workerpool,
+            BigInteger workerpoolmaxprice,
+            String requester,
+            BigInteger volume,
+            String tag,
+            BigInteger category,
+            BigInteger trust,
+            String beneficiary,
+            String callback,
+            String params,
+            String salt,
+            String sign) {
+        super(volume, tag, salt, sign);
+        this.app = toLowerCase(app);
+        this.appmaxprice = appmaxprice;
+        this.dataset = toLowerCase(dataset);
+        this.datasetmaxprice = datasetmaxprice;
+        this.workerpool = toLowerCase(workerpool);
+        this.workerpoolmaxprice = workerpoolmaxprice;
+        this.requester = toLowerCase(requester);
+        this.category = category;
+        this.trust = trust;
+        this.beneficiary = toLowerCase(beneficiary);
+        this.callback = toLowerCase(callback);
+        this.params = params;
+    }
+
+    public boolean equalsExcludedSaltSignAndParams(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        RequestOrder that = (RequestOrder) o;
+        return Objects.equals(app, that.app) &&
+                Objects.equals(appmaxprice, that.appmaxprice) &&
+                Objects.equals(dataset, that.dataset) &&
+                Objects.equals(datasetmaxprice, that.datasetmaxprice) &&
+                Objects.equals(workerpool, that.workerpool) &&
+                Objects.equals(workerpoolmaxprice, that.workerpoolmaxprice) &&
+                Objects.equals(volume, that.volume) &&
+                Objects.equals(category, that.category) &&
+                Objects.equals(trust, that.trust) &&
+                Objects.equals(tag, that.tag) &&
+                Objects.equals(beneficiary, that.beneficiary) &&
+                Objects.equals(callback, that.callback) &&
+                //Objects.equals(params, that.params) &&
```

Comment:
To be consistent, I'd prefer:
1. Either remove that commented out line
2. Add commented out lines for `salt` and `sign`

Reply:
Please see 3534b78 for method removal